### ISSUE TITLE
CMake detecting dependency assumes PACKAGE_LIBRARIES are libraries for one configuration

### ISSUE THREAD

**Issue by gerard-ryan-immersaview** [mesonbuild/meson#9197]
**Describe the bug**
[/mesonbuild/dependencies/cmake.py:451](https://github.com/mesonbuild/meson/blob/0.59.1/mesonbuild/dependencies/cmake.py#L451) assumes that PACKAGE_LIBRARIES are for only one configuration, but [FindOpenSSL.cmake](https://github.com/Kitware/CMake/blob/master/Modules/FindOpenSSL.cmake#L359-L360) for example uses [`select_library_configurations()`](https://cmake.org/cmake/help/latest/module/SelectLibraryConfigurations.html) which creates a list with both debug and release libraries that requires further parsing.

**To Reproduce**
```meson.build
project('hello-world', 'c')
openssl_dep = dependency('openssl')
executable('hello-world', 'main.c', dependencies: openssl_dep)
```
```cmd
meson setup --buildtype=debug build
meson.py compile -C build
```

**Expected behaviour**
The linker will attempt to link with only the debug OpenSSL libraries. 
AKA not both with erroneous keywords like "optimized".
```
"optimized" "C:/Program Files/OpenSSL-Win64/lib/VC/libssl64MD.lib" "debug" "C:/Program Files/OpenSSL-Win64/lib/VC/libssl64MDd.lib" "optimized" "C:/Program Files/OpenSSL-Win64/lib/VC/libcrypto64MD.lib" "debug" "C:/Program Files/OpenSSL-Win64/lib/VC/libcrypto64MDd.lib"
```

**system parameters**
* Is this a [cross build](https://mesonbuild.com/Cross-compilation.html) or just a plain native build (for the same computer)? **plain native**
* what operating system (e.g. MacOS Catalina, Windows 10, CentOS 8.0, Ubuntu 18.04, etc.) **Windows 10**
* what Python version are you using e.g. 3.8.0 **3.9.2**
* what `meson --version` **0.59.1**
* what `ninja --version` if it's a Ninja build **1.10.0.git.kitware.jobserver-1**
* what `cmake --version` **3.20.21032501-MSVC_2**
* what `cl` version **Microsoft (R) C/C++ Optimizing Compiler Version 19.29.30133 for x64**

This seems to be the cause of https://gitlab.freedesktop.org/libnice/libnice/-/issues/140

**Comment by mensinda** [comment]
The best workaround, for now, should be to explicitly let meson know what [imported CMake target](https://github.com/Kitware/CMake/blob/master/Modules/FindOpenSSL.cmake#L594) to use with the [modules kwarg](https://mesonbuild.com/Dependencies.html#cmake).

So, `dependency('OpenSSL', modules: ['OpenSSL::SSL'])` should do the trick here.

**Comment by mensinda** [comment]
Should be fixed with #9221.
